crates/vm/levm/src/opcode_handlers/system.rs

```diff
@@ -458,4 +464,143 @@ impl VM {
+        );
+
+        let new_address = match salt {
+            Some(salt) => Self::calculate_create2_address(sender_address, &code, salt)?,
+            None => Self::calculate_create_address(sender_address, sender_account_info.nonce)?,
+        };
+
+        // 3. Account has nonce or code.
+        if self.get_account(new_address).has_code_or_nonce() {
+            current_call_frame
+                .stack
+                .push(U256::from(CREATE_DEPLOYMENT_FAIL))?;
+            return Ok(OpcodeSuccess::Continue);
+        }
+
+        // THIRD: Changes to the state
+        // 1. Creating contract.
+        let new_account = Account::new(value_in_wei_to_send, code.clone(), 1, Default::default());
+        cache::insert_account(&mut self.cache, new_address, new_account);
+
+        // 2. Increment sender's nonce.
+        self.increment_account_nonce(sender_address)?;
+
+        // 3. Decrease sender's balance.
+        self.decrease_account_balance(sender_address, value_in_wei_to_send)?;
+
+        let max_message_call_gas = max_message_call_gas(current_call_frame)?;
+        let mut new_call_frame = CallFrame::new(
+            sender_address,
+            new_address,
+            new_address,
+            code,
+            value_in_wei_to_send,
+            Bytes::new(),
+            false,
+            max_message_call_gas,
+            U256::zero(),
+            new_depth,
+        );
+
+        self.accrued_substate.created_accounts.insert(new_address); // Mostly for SELFDESTRUCT during initcode.
+        self.accrued_substate.touched_accounts.insert(new_address);
+
+        let tx_report = self.execute(&mut new_call_frame)?;
+
+        current_call_frame.gas_used = current_call_frame
+            .gas_used
+            .checked_add(tx_report.gas_used.into())
+            .ok_or(VMError::OutOfGas(OutOfGasError::ConsumedGasOverflow))?;
+        current_call_frame.logs.extend(tx_report.logs);
+
+        match tx_report.result {
+            TxResult::Success => {
+                // New account's bytecode is going to be the output of initcode exec.
+                self.update_account_bytecode(new_address, tx_report.output)?;
+                current_call_frame
+                    .stack
+                    .push(address_to_word(new_address))?;
+            }
+            TxResult::Revert(_) => {
+                // Return value to sender
+                self.increase_account_balance(sender_address, value_in_wei_to_send)?;
+
+                // Deployment failed so account shouldn't exist
+                cache::remove_account(&mut self.cache, &new_address);
+                self.accrued_substate.created_accounts.remove(&new_address);
+                self.accrued_substate.touched_accounts.remove(&new_address);
+
+                current_call_frame
+                    .stack
+                    .push(U256::from(CREATE_DEPLOYMENT_FAIL))?;
+            }
+        }
+
+        Ok(OpcodeSuccess::Continue)
+    }
```

Comment:
Question, why this "generic create" is in this file and the `generic_call` is in vm.rs?

Reply:
The objective was to declutter `vm.rs`, I plan to move `generic_call` to that file too.